lib/adminSession.js

```diff
@@ -0,0 +1,94 @@
+import { googleCredentials, oauth2URI } from '../config/keys'
+import camelize from 'camelize'
+import 'whatwg-fetch'
+
+export const OAUTH_TOKEN  = 'linksf::oauthToken'
+export const CURRENT_USER = 'linksf::currentUser'
+
+export const authenticate = () => {
+  const user = currentUser()
+
+  if (!user) {
+    window.location = '/login'
+    return false
+  }
+
+  const tokenExpiresAt = new Date(user.tokenExpiresAt*1000)
+  const now = new Date()
+
+  if (tokenExpiresAt < now) {
+    // need to verify token with google
+    const token = sessionStorage[OAUTH_TOKEN]
+    validateGoogleOauthToken(token)
+  } else {
+    // still have a valid token
+    return true
+  }
+}
+
+export const currentUser = () => {
+  try {
+    const user = JSON.parse(sessionStorage[CURRENT_USER])
+
+    if (typeof user == 'object') {
```

Comment:
This is mostly protecting agains javascript+sessionStorage being dumb.  There is chance that someone will try to write `null` to the session storage, which gets converted to a string. Then the `JSON` library tries to be "helpful" and allows you to parse `""` which is technically invalid JSON.  There is some work that needs to be done on this anyway, because there is an edge case I haven't found that will write an empty object there and this check then breaks.